# Helix Antenna Design Guide

- [Step-by-Step Calculation](#step-by-step-calculation)
- [Fast Calculation with Approximations](#fast-calculation-with-approximations)

In [14]:
import numpy as np

---

## Step-by-Step Calculation

The Antenna design process is based on this [paper](https://bpb-us-e1.wpmucdn.com/sites.gatech.edu/dist/4/463/files/2015/06/HelixAPMagazineSubmission.pdf?bid=463). More information can be found [here](https://www.microwaves101.com/encyclopedias/helix-antennas) and [here](https://jcoppens.com/ant/helix/index.en.php).

The paper differentiates between NB (Narrow band) and WB (Wide band) antenna designs. Since we only use a narrow frequency band, we chose an NB antenna for our system, which will be useful to get the needed information via the diagrams from the paper. In addition, manufacturing is based on a prior prototype of an helical antenna.

The first step we took, was researching the frequency of the radio module in our case it is $f=869.525\,\mathrm{MHz}$ and calculate the wavelength with $\lambda = \frac{c}{f} = 345.02\,\mathrm{mm}$. Further more we will use a wire radius of $r = 3\,\mathrm{mm}$ due to good availability. 

In [15]:
f = 869.525e+6
r = 3e-3

λ = 3e+8 / f

print(f'λ:  {λ*1e+2:.3f} \tcm')

λ:  34.502 	cm


### 1. Length and Diameter

The first design step is to specify the circumference $C$ and the length $L$:

<p align="center"><img src="images/helix_plot_1.png" width = 600/></p>

From this diagram we took a ratio of $\frac{L}{C} = 5$ which in our case was a compromise between performance and mechanical instability. With this we got $\frac{C}{\lambda} = 0.9$.

$\Rightarrow C = \lambda \cdot 0.9 = 31.051\,\mathrm{cm}$ 

$\Rightarrow L = 5 \cdot C = 1.553\,\mathrm{m}$

In [16]:
L_C = 5
C_λ = 0.9

C = λ * C_λ
L = C * L_C

print(f'C:  {C*1e+2:.3f} \tcm\nL:  {L*1e+0:.3f} \tm')

C:  31.051 	cm
L:  1.553 	m


### 2. Pitch Angle

Next, the pitch angle has to be specified:

<p align="center"><img src="images/helix_plot_2.png" width = 600/></p>

To get a value for the pitch angle $\alpha$ we first have to calculate:
$$
\frac{r}{C} = 0.00966 \approx 0.01    
$$

In [17]:
r_C = r / C

print(f'r/C:  {r_C:.5f}')

r/C:  0.00966


This values is between $0.0015$ and $0.015$, but is much larger than the first ratio, therefore we chose an angle at $\frac{L}{C} = 5$ which is a little bit lower than $0.015$, which would then be $\alpha = 7.25°$.

In [18]:
α = 7.25

### 3. Remaining Parameters

With the basic parameters we can then calculate the remaining parameters:

$$
d = \tan \alpha \cdot C =39.502\,\mathrm{mm}
$$
$$
N = \frac{L}{d} = 39.303
$$
$$
R= \frac{C}{2\pi} = 49.420\,\mathrm{mm}
$$
$$
l = N \cdot \sqrt{C^2+d^2} = 12.303\,\mathrm{m}
$$

In [19]:
d = np.tan(α * np.pi / 180) * C
N = L / d
R = C / (2 * np.pi)
D = 2 * R
l = N * np.sqrt(C**2 + d**2)

print(f'D:  {D*1e+3:.3f} \tmm\nR:  {R*1e+3:.3f} \tmm\nd:  {d*1e+3:.3f} \tmm\nN:  {N*1e+0:.3f}\nl:  {l*1e+0:.3f} \tm')

D:  98.840 	mm
R:  49.420 	mm
d:  39.502 	mm
N:  39.303
l:  12.303 	m


### 4. Gain

With these informations, we can also estimate the antenna gain:

<p align="center"><img src="images/helix_plot_3.png" width = 600/></p>

We get a gain of $G = 16.5\,\mathrm{dBi}$.

### 5. Summary

The following table summarizes the geometric parameters of the antenna: 

| Parameter | Symbol | Value |
| --- | --- | --- |
| Frequency | f | 869.525 MHz |
| Wavelength | λ | 34.502 cm |
| Circumference | C | 31.051 cm |
| Diameter | D | 98.840 mm |
| Radius | R | 49.420 mm |
| Length | L | 1.553 m |
| Pitch angle | α | 7.25° |
| Pitch distance | d | 39.502 mm |
| Turns | N | 39.303 |
| Wire radius | r | 3 mm |
| Wire length | l | 12.303 m |

In [20]:
print(f'f:  {f*1e-6:.3f} \tMHz\nλ:  {λ*1e+2:.3f} \tcm\nC:  {C*1e+2:.3f} \tcm\nD:  {D*1e+3:.3f} \tmm\nR:  {R*1e+3:.3f} \tmm\nL:  {L*1e+0:.3f} \tm\nα:  {α*1e+0:.3f} \t°\nd:  {d*1e+3:.3f} \tmm\nN:  {N*1e+0:.3f}\nr:  {r*1e+3:.0f} \t\tmm\nl:  {l*1e+0:.3f} \tm')

f:  869.525 	MHz
λ:  34.502 	cm
C:  31.051 	cm
D:  98.840 	mm
R:  49.420 	mm
L:  1.553 	m
α:  7.250 	°
d:  39.502 	mm
N:  39.303
r:  3 		mm
l:  12.303 	m


---

## Fast Calculation with Approximations

These calculations are also based on the step-by-step design process according to the paper.
We approximate the diagrams so you only need to specify frequency $f$, wire radius $r$ and ratio of length to circumference $\frac{L}{C}$.

In [21]:
### User Input

f = 869.525e+6
r = 3e-3
L_C = 5

### Calculations

λ = 3e+8 / f
C_λ = np.interp(np.log10(L_C), [np.log10(0.7), np.log10(20)], [1.025, 0.83])

C = λ * C_λ
L = C * L_C

r_C = r / C
α = np.interp(np.log10(L_C), [np.log10(0.7), np.log10(20)], [5.5, 9.5]) + (np.log10(r_C) - np.log10(0.015)) * 1.5

d = np.tan(α * np.pi / 180) * C
N = L / d
R = C / (2 * np.pi)
D = 2 * R
l = N * np.sqrt(C**2 + d**2)

G = np.interp(np.log10(L_C), [np.log10(0.7), np.log10(20)], [12, 20])

### Output

print(f'f:  {f*1e-6:.3f} \tMHz\nλ:  {λ*1e+2:.3f} \tcm\nC:  {C*1e+2:.3f} \tcm\nD:  {D*1e+3:.3f} \tmm\nR:  {R*1e+3:.3f} \tmm\nL:  {L*1e+0:.3f} \tm\nα:  {α*1e+0:.3f} \t°\nd:  {d*1e+3:.3f} \tmm\nN:  {N*1e+0:.3f}\nr:  {r*1e+3:.0f} \t\tmm\nl:  {l*1e+0:.3f} \tm\n\nG:  {G*1e+0:.3f} \tdBi')

f:  869.525 	MHz
λ:  34.502 	cm
C:  31.418 	cm
D:  100.008 	mm
R:  50.004 	mm
L:  1.571 	m
α:  7.552 	°
d:  41.651 	mm
N:  37.716
r:  3 		mm
l:  11.953 	m

G:  16.692 	dBi
